plugins/jobtype/src/azkaban/jobtype/HadoopSparkJob.java

```diff
@@ -0,0 +1,422 @@
+    argList.add("--" + QUEUE);
+    argList.add(jobProps.getString(QUEUE, QUEUE_DEFAULT));
+    argList.add("--" + DRIVER_MEMORY_SPARK);
+    argList.add(jobProps.getString(DRIVER_MEMORY_AZKABAN, DRIVER_MEMORY_DEFAULT));
+    argList.add("--" + EXECUTOR_MEMORY_SPARK);
+    argList.add(jobProps.getString(EXECUTOR_MEMORY_AZKABAN, EXECUTOR_MEMORY_DEFAULT));
+    // always place EXECUTION_JAR_AZKABAN in the last spot
+    argList.add(jobProps.getString(EXECUTION_JAR_AZKABAN));
+
+    return StringUtils.join((Collection<String>) argList, " ");
+  }
+
+  @Override
+  protected List<String> getClassPaths() {
+
+    List<String> classPath = super.getClassPaths();
+
+    classPath.add(getSourcePathFromClass(Props.class));
+    classPath.add(getSourcePathFromClass(HadoopSecureHiveWrapper.class));
+    classPath.add(getSourcePathFromClass(HadoopSecurityManager.class));
+
+    classPath.add(HadoopConfigurationInjector.getPath(getJobProps(), getWorkingDirectory()));
+    List<String> typeClassPath = getSysProps().getStringList("jobtype.classpath", null, ",");
+    if (typeClassPath != null) {
+      // fill in this when load this jobtype
+      String pluginDir = getSysProps().get("plugin.dir");
+      for (String jar : typeClassPath) {
+        File jarFile = new File(jar);
+        if (!jarFile.isAbsolute()) {
+          jarFile = new File(pluginDir + File.separatorChar + jar);
+        }
+
+        if (!classPath.contains(jarFile.getAbsoluteFile())) {
+          classPath.add(jarFile.getAbsolutePath());
+        }
+      }
+    }
+
+    List<String> typeGlobalClassPath = getSysProps().getStringList("jobtype.global.classpath",
+            null, ",");
+    if (typeGlobalClassPath != null) {
+      for (String jar : typeGlobalClassPath) {
+        if (!classPath.contains(jar)) {
+          classPath.add(jar);
+        }
+      }
+    }
+
+    return classPath;
+  }
+
+  private static String getSourcePathFromClass(Class<?> containedClass) {
+    File file = new File(containedClass.getProtectionDomain().getCodeSource().getLocation()
+            .getPath());
+
+    if (!file.isDirectory() && file.getName().endsWith(".class")) {
+      String name = containedClass.getName();
+      StringTokenizer tokenizer = new StringTokenizer(name, ".");
+      while (tokenizer.hasMoreTokens()) {
+        tokenizer.nextElement();
+        file = file.getParentFile();
+      }
+
+      return file.getPath();
+    } else {
+      return containedClass.getProtectionDomain().getCodeSource().getLocation().getPath();
+    }
+  }
+
+  /**
+   * Resolves /* and expands them into appropriate jar file names. Only works for jar file as of now
+   * (only known use case)
+   * 
+   * test case required: one replacement, two replacement.
+   * 
+   * @param unresolvedJar
+   * @return jar file list, comma separated, all .../* expanded into actual jar names in order
+   */
+  private String resolveJars(String unresolvedJar) {
```

Comment:
Should we move it to utils ? It may be useful in other places in future plus that will help use write unit tests for these method ?